command/agent/config_parse_test.go

```diff
@@ -682,3 +416,89 @@ func removeHelperAttributes(c *Config) *Config {
 	}
 	return c
 }
+
+func (c *Config) addDefaults() {
+	if c.Client.ServerJoin == nil {
+		c.Client.ServerJoin = &ServerJoin{}
+	}
+	if c.ACL == nil {
+		c.ACL = &ACLConfig{}
+	}
+	if c.Consul == nil {
+		c.Consul = config.DefaultConsulConfig()
+	}
+	if c.Autopilot == nil {
+		c.Autopilot = config.DefaultAutopilotConfig()
+	}
+	if c.Vault == nil {
+		c.Vault = config.DefaultVaultConfig()
+	}
+	if c.Telemetry == nil {
+		c.Telemetry = &Telemetry{}
+	}
+	if c.Server == nil {
+		c.Server = &ServerConfig{}
+	}
+	if c.Server.ServerJoin == nil {
+		c.Server.ServerJoin = &ServerJoin{}
+	}
+}
+
+// this tests for a panic parsing json with an object of exactly
+// length 1 described in
+// https://github.com/hashicorp/nomad/issues/1290
+func TestConfig_ParsePanic(t *testing.T) {
+	c, err := ParseConfigFile("./testdata/obj-len-one.hcl")
+	if err != nil {
+		t.Fatalf("parse error: %s\n", err)
+	}
+
+	d, err := ParseConfigFile("./testdata/obj-len-one.json")
+	if err != nil {
+		t.Fatalf("parse error: %s\n", err)
+	}
+
+	require.EqualValues(t, c, d)
+}
+
+// this tests for top level keys left by hcl when parsing slices in
+// the config structure
+func TestConfig_ParseSliceExtra(t *testing.T) {
```

Comment:
nice tests

Reply:
@preetapan I added an additional sample test to reflect some more real world json configuration files